.github/workflows/publish.yml

```diff
@@ -0,0 +1,39 @@
+name: Build and Publish
+
+on:
+  push:
+    tags:
+    - '*'
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  test_wheel_building:
+    permissions:
+      contents: none
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/publish_pure_python.yml@v1
+    with:
+      upload_to_pypi: false
+      upload_to_anaconda: false
+      test_extras: test
+      test_command: pytest $GITHUB_WORKSPACE/tests $GITHUB_WORKSPACE/docs -p no:warnings
+
+  build_and_publish:
```

Comment:
This seems to build everything twice? Which is going to be expensive once we switch on the compilation... The usual way of doing this in steps is

1. Build sdist & store artefacts,
2. Build wheels & store artefacts,
3. Upload artefacts if publishing.

An example of this is again [here](https://github.com/ntessore/healpix/blob/c3d421ca8fc02d737cf973b69176eec913dd0e31/.github/workflows/cibuildwheel.yml).

Reply:
Done!